service/controller/resource/certificates/resource.go

```diff
@@ -103,16 +102,22 @@ func (r *Resource) getDesiredObject(v interface{}) (*corev1.Secret, error) {
 	secretData := sourceSecret.Data
 
 	if key.IsCAPICluster(cluster) {
-		// CAPI Secret is a kubeconfig so we need to extract the certs from it
+		// CAPI Secret is a kubeconfig so we need to extract the certs or tokens from it
 		if kubeconfig, ok := secretData["value"]; ok {
 			capiKubeconfig, err := clientcmd.Load(kubeconfig)
 			if err != nil {
 				return nil, microerror.Mask(err)
 			}
 			kubeconfigAdminUser := fmt.Sprintf("%s-admin", cluster.GetName())
+			kubeconfigCapiAdminUser := fmt.Sprintf("%s-capi-admin", cluster.GetName())
 			secretData["ca"] = capiKubeconfig.Clusters[cluster.GetName()].CertificateAuthorityData
-			secretData["crt"] = capiKubeconfig.AuthInfos[kubeconfigAdminUser].ClientCertificateData
-			secretData["key"] = capiKubeconfig.AuthInfos[kubeconfigAdminUser].ClientKeyData
+			if _, ok := capiKubeconfig.AuthInfos[kubeconfigAdminUser]; ok {
+				secretData["crt"] = capiKubeconfig.AuthInfos[kubeconfigAdminUser].ClientCertificateData
+				secretData["key"] = capiKubeconfig.AuthInfos[kubeconfigAdminUser].ClientKeyData
+			} else if _, ok := capiKubeconfig.AuthInfos[kubeconfigCapiAdminUser]; ok {
+				// This is the case for eks clusters.
+				secretData["token"] = []byte(capiKubeconfig.AuthInfos[kubeconfigCapiAdminUser].Token)
+			}
```

Comment:
If feel like we should return an error in case none of the keys exist.

Reply:
I feel like you're right :)